### PR TITLE
Nb view config list

### LINKED ISSUE #1147: Implement NodeBalancer view summary
Summary Section here:
https://projects.invisionapp.com/share/8FALUJ0EY#/screens/220575688

### LINKED ISSUE #1311: Implement NodeBalancer view config list
Config List here:
https://projects.invisionapp.com/share/8FALUJ0EY#/screens/220575688

### PR DESCRIPTION
**Pull request by abemassry**
closes #1147
closes #1311

Nodebalancer view config list works, pulls data from the API tested
manually, automatically, tests pass.

Still need to work on summary.

### CODE REVIEW

**Comment by abemassry** [comment]
Work on summary

**Comment by abemassry** [comment]
Add screenshot

File: src/nodebalancers/nodebalancer/layouts/IndexPage.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  constructor(props) {
+    super(props);
+    const updateCurrentNodebalancer = (setState) => (props) => {
+      const currentNodebalancer = Object.values(props.nodebalancers.nodebalancers).filter(
+        n => n.label === props.params.nbLabel)[0];
+      (setState || this.setState)({ currentNodebalancer });
+    };
+
+    // currentNodebalancer needs to be updated now and every time props changes
+    this.componentWillUpdate = updateCurrentNodebalancer;
```
Comment (mparke) [inline src/nodebalancers/nodebalancer/layouts/IndexPage.js]: this changes the interface of this method, see what @eatonphil did here: https://github.com/linode/manager/pull/1357/files#diff-2babf68a139324c8915124d37e0dc600R21

Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/IndexPage.js]: Here's a better place to look: https://github.com/linode/manager/pull/1366/files#diff-2babf68a139324c8915124d37e0dc600R24

Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/IndexPage.js]: (pending discussion / approval)

File: src/nodebalancers/nodebalancer/layouts/IndexPage.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+              </div>
+            </div>
+            <div className="row">
+              <div className="col-sm-1 row-label">
+                Status
+              </div>
+              <div className="col-sm-9">
+                {nodebalancer.status
+                  .replace(/_/, ' ')
+                  .replace(/[a-z]/, (l) => l.toUpperCase())
```
Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/IndexPage.js]: I'd prefer to see this as a dictionary lookup similar to [Linode states](https://github.com/linode/manager/blob/master/src/constants.js#L30).

File: src/nodebalancers/nodebalancer/layouts/IndexPage.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        <div className="container">
+          <Card title="Summary">
+            <div className="row">
+              <div className="col-sm-1 row-label">
+                IP Addresses
+              </div>
+              <div className="col-sm-9">
+                <ul className="list-unstyled">
+                  <li>{nodebalancer.ipv4}</li>
+                  <li className="text-muted">{nodebalancer.ipv6}</li>
```
Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/IndexPage.js]: +1 to text-muted

File: src/nodebalancers/nodebalancer/layouts/IndexPage.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            </h1>
+          </div>
+        </header>
+        <div className="container">
+          <Card title="Summary">
+            <div className="row">
+              <div className="col-sm-1 row-label">
+                IP Addresses
+              </div>
+              <div className="col-sm-9">
```
Comment (eatonphil) [inline src/nodebalancers/nodebalancer/layouts/IndexPage.js]: Could you replace this with sm-11?

**Comment by eatonphil** [comment]
Looks good after requested changes. Will test locally by end of day.

**Pull request merged**